sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -784,6 +787,25 @@ public Node visitPrimaryKeyConstraint(SqlBaseParser.PrimaryKeyConstraintContext
         return new PrimaryKeyConstraint<>(visitCollection(context.columns().primaryExpression(), Expression.class));
     }
 
+    @Override
+    public Node visitTableCheckConstraint(SqlBaseParser.TableCheckConstraintContext context) {
```

Comment:
you can probably even use the same rule name '#columnCheckConstraint' in grammar for both cases and implement only one visitor

Reply:
In which case I would get:

```
Task :sql-parser:generateGrammarSource FAILED
error(123): /Users/marregui/workspace/crate/sql-parser/src/main/antlr/SqlBase.g4:520:86: rule alt label columnCheckConstraint redefined in rule tableElement, originally in rule columnConstraint
```

Reply:
ah, never mind, also they have different types